crypto/ec/ec_curve.c

```diff
@@ -3097,6 +3098,32 @@ static EC_GROUP *ec_group_new_from_data(const ec_list_element curve)
             goto err;
         }
     }
+
+    if (EC_GROUP_get_asn1_flag(group) == OPENSSL_EC_NAMED_CURVE) {
+        /*
+         * Some curves don't have an associated OID: for those we should not
+         * default to `OPENSSL_EC_NAMED_CURVE` encoding of parameters and
+         * instead set the ASN1 flag to `OPENSSL_EC_EXPLICIT_CURVE`.
+         *
+         * Note that `OPENSSL_EC_NAMED_CURVE` is set as the default ASN1 flag on
+         * `EC_GROUP_new()`, when we don't have enough elements to determine if
+         * an OID for the curve name actually exists.
+         * We could implement this check on `EC_GROUP_set_curve_name()` but
+         * overloading the simple setter with this lookup could have a negative
+         * performance impact and unexpected consequences.
+         */
+        ASN1_OBJECT *asn1obj = OBJ_nid2obj(curve.nid);
```

Comment:
The careful reviewer would notice that this check is superfluous (as hinted in the comment), given that `EC_GROUP_new()` will set the asn1 flag to `OPENSSL_EC_NAMED_CURVE`: the reason why I have a fixup commit styling it this way intentionally is that I want to have a local scope for declaring `ASN1_OBJECT *asn1obj` and to handle its life-cycle.

In `1.1.1` this is superfluous --but still looks tidy to me-- but in the perspective of forward-porting this to 3.0.0 it makes more sense as likely this whole section will be disabled when `FIPS_MODE` is defined, and having this local scope prevents ugly `#ifdef` spaghettization (and as an Italian :it: I am sensitive about spaghetti :spaghetti: :stuck_out_tongue_closed_eyes:).